src/tuner/nccl_ofi_regions.cpp

```diff
@@ -1167,6 +1169,80 @@ static ncclResult_t region_init_internal_p6(nccl_ofi_tuner_region_context_t *reg
 }
 
 
+static uint64_t calculateChunkSizeTreeLL128(uint64_t message_size, int nChannels, int log2_nnodes)
+{
+	const int ppn = 8;
+	/* Initial Buffer Size */
+	uint64_t buffSize = NCCL_OFI_TUNER_NCCL_LL128_ELEMS_PER_THREAD * 
+		NCCL_OFI_TUNER_NCCL_LL128_MAX_NTHREADS * 
+		NCCL_OFI_TUNER_NCCL_STEPS * sizeof(uint64_t);
+
+	uint64_t stepSize = buffSize / NCCL_OFI_TUNER_NCCL_STEPS;
+	uint64_t chunkSize = stepSize;
+
+	/* Adjust for Protocol Overhead */
+	chunkSize = (chunkSize / NCCL_OFI_TUNER_NCCL_LL128_LINEELEMS) * NCCL_OFI_TUNER_NCCL_LL128_DATAELEMS;
+
+	/* Estimate the number of communication steps needed for the tree algorithm,
+	 * based on the logarithmic nature of tree reduction (log2 of nodes) plus a 
+	 * small factor for processes per node (ppn) */
+	double nStepsLL128 = 1 + log2_nnodes + 0.1 * ppn;
```

Comment:
not quite as bad as the log2() call, but why do we need to do all this floating point math, just to convert back to integer in every use of nStepsLL128?

Reply:
ok, not thrilled with that, but let's move on.